isis/src/hayabusa2/apps/hyb2onccal/Hyb2OncCalUtils.h

```diff
@@ -114,381 +115,277 @@ static double g_J(1.0);
-    // If we have only one input cube, that means that we do not have a flat-field (W1/W2).
-    if (in.size() == 2) {
-      // Note that our current flat-fields to not have special pixel values.
-      if (IsSpecial(flatField[i]) || IsSpecial(imageOut[i]))
-      {
-        imageOut[i] = Isis::Null;
+      // Check for special pixel in input image and pass through
+      if ( IsSpecial(imageOut[i]) ) {
+        imageOut[i] = imageIn[i];
         continue;
       }
-      else {
-        if (flatField[i] != 0) {
-          imageOut[i] /= (flatField[i]);
-        }
-      }
-    }
-  }
-  return;
-}
 
+      // Apply compression factor here to raise LOSSY dns to proper response
 
-/**
- * @brief Translates a 1-banded Isis::Cube to an OpenMat object
- *
- * @author 2016-04-19 Tyler Wilson
- *
- * @param icube A pointer to the input cube
- *
- * @return @b Mat A pointer to the OpenMat object
- */
-Mat *isis2mat(Cube *icube) {
-  int nlines = icube->lineCount();
-  int nsamples = icube->sampleCount();
-  Mat *matrix = new Mat(nlines, nsamples, CV_64F);
-
-  // Set up line manager and read in the data
-  LineManager linereader(*icube);
-  for (int line = 0; line < nlines; line++) {
-    linereader.SetLine(line + 1);
-    icube->read(linereader);
-    for (int samp = 0;  samp < nsamples; samp++) {
-      matrix->at<double>(line, samp) = (double)linereader[samp];
-    }
- }
-return matrix;
-}
-
-
-/**
- * @brief Translates an OpenMat object to an ISIS::Cube with one band
- *
- * @author 2016-04-19 Tyler Wilson
- *
- * @param matrix A pointer to the OpenMat object
- *
- * @param cubeName The name of the Isis::Cube that is being created.
- */
-void mat2isis(Mat *matrix, QString cubeName) {
-
-  int nlines = matrix->rows;
-  int nsamples = matrix->cols;
-  CubeAttributeOutput set;
-  set.setPixelType(Real);
 
-  Cube ocube;
-  ocube.setDimensions(nsamples, nlines,1);
-  ocube.create(cubeName, set);
+      // BIAS Removal - Only needed if not on-board corrected
+      if (!g_onBoardSmearCorrection) {
+        if ( (imageOut[i] - g_bias) <= 0.0) {
+          imageOut[i] = Null;
+          continue;
+        }
+        else {
 
-  LineManager linewriter(ocube);
+          imageOut[i] = imageOut[i] - g_bias;
+        }
+      }
 
-  for (int line = 0; line < nlines; line++) {
-    linewriter.SetLine(line+1);
+      // Smear correction
+      if (!g_onBoardSmearCorrection) {
+        imageOut[i] = imageOut[i] - smear;
+      }
 
-    for ( int samp=0; samp < nsamples; samp++ ) {
+      // Both dark current and linearity correction are commented out for now since the JAXA team
+      // does not currently do these steps.
+
+      // DARK Current
+      // imageOut[i] = imageOut[i] - g_darkCurrent;    
+      
+      // Linearity Correction
+      // double dn = imageOut[i];    
+      // double result = 1.0;
+      // double x0 = 1.0;
+      // newton_rapheson(dn,x0, g_L,result );   
+      // imageOut[i] = result;
```

Comment:
Should this be deleted?

Reply:
The corrections are due to be added later this calendar year. At that time we should be able to uncomment these and rematch the teams output